src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -138,12 +157,24 @@ void KeyboardManagerState::UpdateDetectShortcutUI()
     }
 
     std::unique_lock<std::mutex> detectedShortcut_lock(detectedShortcut_mutex);
-    hstring shortcutString = convertVectorToHstring<DWORD>(detectedShortcut);
+
+    std::vector<hstring> shortcutKeys(detectedShortcut.size());
+    std::transform(
+        detectedShortcut.begin(), 
+        detectedShortcut.end(), 
+        shortcutKeys.begin(), 
+        [](DWORD k) { return winrt::to_hstring((unsigned)k); }
+    );
     detectedShortcut_lock.unlock();
 
     // Since this function is invoked from the back-end thread, in order to update the UI the dispatcher must be used.
-    currentShortcutTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [=]() {
-        currentShortcutTextBlock.Text(shortcutString);
+    currentShortcutTextBlock.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [this, shortcutKeys]() {

```

Comment:
[nit] we can probably name this to something better, now that it is no longer a textblock. Maybe `currentDetectShortcutUI` or something along those lines? Similar for the single key as well.

Reply:
Good catch, I think `currentShortcutUI` would be an appropiate name. The same with `currentSingleKeyUI`